eval_test.go

```diff
@@ -189,37 +214,124 @@ func TestSimplifyRepo(t *testing.T) {
 	}
 }
 
+func hash(name string) uint32 {
+	h := fnv.New32()
+	h.Write([]byte(name))
+	return h.Sum32()
+}
+
 func TestSimplifyRepoBranch(t *testing.T) {
 	d := compoundReposShard(t, "foo", "bar")
 
-	some := &query.RepoBranches{Set: map[string][]string{"bar": {"branch1"}}}
-	none := &query.Repo{"banana"}
+	none := &query.Repo{Pattern: "banana"}
+	for _, tc := range []struct {
```

Comment:
I could re-work tests in this PR to add an independent test rather than refactoring them into table driven tests which makes the diff harder to read.